src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
Good point. One solution would be to modify [extractFeatures](https://github.com/LARG/HFO/blob/master/src/feature_extractor.h#L15) to be:

virtual const std::vector<float>& ExtractFeatures(const rcsc::WorldModel& wm, boolean last_action_status) = 0;

This way agent.cpp could pass the correct last action status. Sample player could pass something default such as always 1.